### PR TITLE
Create erreurs medicamenteuses tables in db

### PR DESCRIPTION
**Pull request by lrahal**

### CODE REVIEW

File: create_database/main.py [hunk 1]
```diff
@@ -73,6 +78,67 @@ def load_to_df_atc(fpath):
...
+    if fpath.exists():
+        df = load_to_df_emed(fpath, _settings)
+        df["produit_denom"] = df.denomination.apply(em.get_produit_denom)
+        df["forme_denom"] = df.denomination.apply(em.get_forme_denom)
+
+        df_spe = pd.read_sql("specialite", conn)
+        df_spe = df_spe.set_index("cis")
+
+        df_corresp = em.get_corresp_df(df, df_spe)
+        args = {**{"name": "cis_erreurs_med_corresp"}, **_settings["to_sql"]}
```
Comment (jmunozz) [inline create_database/main.py:128]: here add index type for sql

File: create_database/main.py [hunk 2]
```diff
@@ -73,6 +78,67 @@ def load_to_df_atc(fpath):
...
+        db.create_table_from_df(df_corresp, args)
+
+        for table_name, table_column in tqdm(_settings["tables"].items()):
+            print("{} table creation".format(table_name))
+            df_table = em.get_table_df(df, df_spe, table_column)
+            args = {
```
Comment (jmunozz) [inline create_database/main.py:134]: here add index type for sql

**Review by jmunozz — CHANGES_REQUESTED** [review]

File: create_database/erreurs_med.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        (df.produit_denom == produit_specialite) & (df.forme_denom == forme_specialite)
+    ]
+
+
+def get_erreur_df(df_base: pd.DataFrame, cis: str, col: str) -> pd.DataFrame:
+    df = df_base.groupby(col).denomination.count().reset_index()
+    df["pourcentage"] = df.apply(
+        lambda x: round(x.denomination / df.denomination.sum() * 100, 2), axis=1
+    )
+    df = df.drop(columns=["denomination"])
```
[the remark this answers is not in the document]
Comment (lrahal) [reply create_database/erreurs_med.py:41]: Sublime

File: create_database/main.py [hunk 4]
```diff
@@ -21,8 +26,8 @@ def download_bdpm_cis():
     return bdpm_cis_path
 
 
-def load_to_df_bdpm_cis(path, settings):
-    args = {**{"filepath_or_buffer": path}, **settings}
+def load_to_df_bdpm_cis(fpath, settings):
```
[the remark this answers is not in the document]
Comment (lrahal) [reply create_database/main.py:29]: Okay but this function was coded outside this branch. Let's merge and do the change afterwards :)

File: create_database/settings.py [hunk 5]
```diff
@@ -101,4 +100,49 @@
             "dtype": {"code_atc": String(16)},
         },
     },
+    "erreurs_med": {
+        "source": {"pattern": "RqHackathon_20190911.xlsx"},
+        "read_excel": {
```
[the remark this answers is not in the document]
Comment (lrahal) [reply create_database/settings.py:105]: What would be the index in this case ? denomination ?

File: create_database/main.py [hunk 6]
```diff
@@ -73,6 +78,67 @@ def load_to_df_atc(fpath):
...
+        lambda x: "Non renseigné" if x == "NI" else x
+    )
+    return df
+
+
+def create_table_emed(_settings):
+    fpath = find_emed(_settings["source"]["pattern"])
+    if fpath.exists():
+        df = load_to_df_emed(fpath, _settings)
+        df["produit_denom"] = df.denomination.apply(em.get_produit_denom)
```
[the remark this answers is not in the document]
Comment (lrahal) [reply create_database/main.py:121]: Here where ??????????????????
??????????
?????
???
?

File: create_database/main.py [hunk 7]
```diff
@@ -73,6 +78,67 @@ def load_to_df_atc(fpath):
...
+        df["forme_denom"] = df.denomination.apply(em.get_forme_denom)
+
+        df_spe = pd.read_sql("specialite", conn)
```
[the remark this answers is not in the document]
Comment (lrahal) [reply create_database/main.py:124]: Mmmmmmh pas compris 🍡

File: create_database/settings.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (jmunozz) [reply create_database/settings.py:105]: the index is something that you want to be searchable

**Pull request merged**